test/integration_test/migration_dr_actions_test.go

```diff
@@ -515,3 +800,83 @@ func waitTillActionComplete(t *testing.T, actionType storkv1.ActionType, actionN
 	}
 	log.FailOnError(t, fmt.Errorf("DR action %s did not complete in 15 minutes", actionName), "DR action did not complete in 15 minutes")
 }
+
+func createSchedulePolicyWithMockedTime(t *testing.T, policyName string, policyType string) time.Time {
+	retain := 5
+	intervalMinutes := 15
+	// Set first trigger 2 minutes from now
+	scheduledTime := time.Now().Add(2 * time.Minute)
+
+	var nextScheduledTime time.Time
+	var schedPolicy *storkv1.SchedulePolicy
+	switch policyType {
+	case string(storkv1.SchedulePolicyTypeInterval):
+		nextScheduledTime = scheduledTime.Add(time.Duration(intervalMinutes) * time.Minute)
+		schedPolicy = &storkv1.SchedulePolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: policyName,
+			},
+			Policy: storkv1.SchedulePolicyItem{
+				Interval: &storkv1.IntervalPolicy{
+					Retain:          storkv1.Retain(retain),
+					IntervalMinutes: 15,
+				},
+			}}
+	case string(storkv1.SchedulePolicyTypeDaily):
+		nextScheduledTime = scheduledTime.AddDate(0, 0, 1)
+		schedPolicy = &storkv1.SchedulePolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: policyName,
+			},
+			Policy: storkv1.SchedulePolicyItem{
+				Daily: &storkv1.DailyPolicy{
+					Retain: storkv1.Retain(retain),
+					Time:   scheduledTime.Format(time.Kitchen),
+				},
+			}}
+	case string(storkv1.SchedulePolicyTypeWeekly):
+		// Set first trigger 2 minutes from now
+		nextScheduledTime = scheduledTime.AddDate(0, 0, 7)
+		schedPolicy = &storkv1.SchedulePolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: policyName,
+			},
+			Policy: storkv1.SchedulePolicyItem{
+				Weekly: &storkv1.WeeklyPolicy{
+					Retain: storkv1.Retain(retain),
+					Day:    scheduledTime.Weekday().String(),
+					Time:   scheduledTime.Format(time.Kitchen),
+				},
+			}}
+	case string(storkv1.SchedulePolicyTypeMonthly):
+		nextScheduledTime = scheduledTime.AddDate(0, 1, 0)
+		// Set the time to zero in case the date doesn't exist in the next month
+		if nextScheduledTime.Day() != scheduledTime.Day() {
+			nextScheduledTime = time.Time{}
+		}
+		schedPolicy = &storkv1.SchedulePolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: policyName,
+			},
+			Policy: storkv1.SchedulePolicyItem{
+				Monthly: &storkv1.MonthlyPolicy{
+					Retain: storkv1.Retain(retain),
+					Date:   scheduledTime.Day(),
+					Time:   scheduledTime.Format(time.Kitchen),
+				},
+			},
+		}
+	default:
+		log.FailOnError(t, fmt.Errorf("invalid policy type %s", policyType), "Invalid policy type")
+	}
+	if authTokenConfigMap != "" {
+		err := addSecurityAnnotation(schedPolicy)
+		log.FailOnError(t, err, "error updating security annotations for schedule policy %s of type %v", policyName, policyType)
+	}
+
+	_, err := storkops.Instance().CreateSchedulePolicy(schedPolicy)
+	log.FailOnError(t, err, "error creating monthly schedule policy")
```

Comment:
nit.. log statement is hardcoded with monthly type.

Reply:
Done